### PR TITLE
refs #72 #73 Accept typeless _id in constructor

### PR DESCRIPTION
**Pull request by louischan-oursky**

### CODE REVIEW

**Comment by oursky-travis** [comment]
@louischan-oursky, thanks for your PR! By analyzing the annotation information on this pull request, we identified @rickmak to be a potential reviewer

File: lib/record.js [hunk 1]
```diff
@@ -92,11 +92,20 @@ export default class Record {
...
       id = uuid.v4();
     } else {
-      let [type, name] = Record.parseID(id);
-      if (type !== this._recordType) {
-        throw new Error('_id is not valid. RecordType mismatch.');
+      let tuple = null;
+      try {
+        tuple = Record.parseID(id);
+      } catch (e) {
+        // id is typeless
```
Comment (cheungpat) [inline lib/record.js]: Having an exception when parsing the ID doesn’t necessarily imply the string has no type information.

File: lib/record.js [hunk 2]
```diff
@@ -92,11 +92,20 @@ export default class Record {
...
+        // default to recordType
+      }
+      if (tuple) {
+        let [type, name] = tuple;
+        if (type !== this._recordType) {
```
Comment (cheungpat) [inline lib/record.js]: Perhaps you also want to check if `_recordType` exists, and that it equals `this._recordType`.

File: lib/record.js (hunk 1 above)
Comment (cheungpat) [inline lib/record.js]: I wouldn’t say no if you don’t call `parseID` when both `_recordType` field and `_id` exists.

File: test/record.js [hunk 3]
```diff
@@ -85,6 +85,34 @@ describe('Extended Record', function () {
...
+  it('reject _id with different type', function () {
+    expect(function () {
+      let r = new Note({
+        _id: 'box/2'
+      });
+    }).to.throw(
+      '_id is not valid. RecordType mismatch.'
+    );
+  });
+
```
Comment (cheungpat) [inline test/record.js]: You need test case here for the cases:

1. When `_recordType` matches
2. When `_recordType` does not matches

**Comment by cheungpat** [comment]
I wish @louischan-oursky could step up and make changes to the PR. We could all benefit from this PR here.

**Comment by rickmak** [comment]
Superset by  #95

**Pull request closed without merging**